bin/checkVersion.js

```diff
@@ -1,41 +1,100 @@
+/*To avoid dependencies, the compareVersions code below has been copied from: https://github.com/omichelsen/compare-versions*/
 /*eslint no-console: "off"*/
+/*eslint no-useless-escape: "off"*/
+/*eslint indent: "off"*/
 const process = require('process');
 const config = require('../package.json');
-const comparator = config.engines.node.match(/[^0-9.]+/).toString();
-const minVersion = process.version.match(/[0-9.]+/).toString();
-const version = config.engines.node.match(/[0-9.]+/).toString();
-
-function compareVersions(left, right) {
-  if (left === right) return 0;
-  const a = left.split('.').map(Number);
-  const b = right.split('.').map(Number);
-  for (let i = 0; i < a.length; i++) {
-    if (a[i] > b[i]) {
-      return 1;
-    }
-    if (a[i] < b[i]) {
-      return -1;
-    }
-  }
-  return true;
-}
+const version = process.version.match(/[0-9.]+/).toString();
+const minVersion = config.engines.node.match(/[0-9.]+/).toString();
+const semver = /^v?(?:\d+)(\.(?:[x*]|\d+)(\.(?:[x*]|\d+)(\.(?:[x*]|\d+))?(?:-[\da-z\-]+(?:\.[\da-z\-]+)*)?(?:\+[\da-z\-]+(?:\.[\da-z\-]+)*)?)?)?$/i;
+
+const comparators = [
+  { val: '>', validate: r => r == 1 },
+  { val: '>=', validate: r => r == 0 || r == 1 }
```

Comment:
I don't think we care to limit this to just these to comparators, but this is fine for now

Reply:
Ok, I can go ahead and add the other ones in. It won't be much more effort

Reply:
Actually, I don't see this code handling the logic for `^` or `~` semver options so this is good

Reply:
Ok